packages/gateway/src/index.js

```diff
@@ -0,0 +1,43 @@
+import { addCorsHeaders } from './cors.js'
+import { errorHandler } from './error-handler.js'
+import { getCidFromSubdomainUrl } from './utils/cid.js'
+
+/**
+ * @typedef {Object} Env
+ * @property {string} IPFS_GATEWAY
+ */
+
+/**
+ * Handle gateway request
+ * @param {Request} request
+ * @param {Env} env
+ */
+async function handleRequest(request, env) {
+  const publicGatewayUrl = new URL('ipfs', env.IPFS_GATEWAY)
```

Comment:
Can be re-used for every request.

Reply:
It would be nice, but can it? `IPFS_GATEWAYS` are dynamically set and the env context is only accessible on request handler. Any way you aware?